### ISSUE TITLE
Commands not found iPadOS 16.1 iPad Air 5th Gen M1

### ISSUE THREAD

**Issue by arcane58** [holzschu/a-shell#500]
When I run a-shell it seems to be pot luck which commands will work and which will not.

Commands that do not work give the following output or similar:
Failed loading python from python3_ios.framework/python3_ios, cause = dlopen(python3_ios.framework/python3_ios, 0x0009): tried: 'python3_ios.framewo
rk/python3_ios' (relative path not allowed in hardened program), '/private/preboot/Cryptexes/OSpython3_ios.framework/python3_ios' (errno=2), '/priva
te/var/containers/Bundle/Application/F35078DB-6337-4339-AF5B-3A1FA5501A71/a-Shell.app/Frameworks/python3_ios.framework/python3_ios' (could not regis
ter fairplay decryption, mremap_encrypted() => -1), '/private/var/containers/Bundle/Application/F35078DB-6337-4339-AF5B-3A1FA5501A71/a-Shell.app/Fra
meworks/ios_system.framework/Frameworks/python3_ios.framework/python3_ios' (errno=2), '/usr/lib/swift/python3_ios.framework/python3_ios' (errno=2, n
ot in dyld cache), '/private/preboot/Cryptexes/OS/usr/lib/swift/python3_ios.framework/python3_ios' (errno=2), '/private/var/containers/Bundle/Applic
ation/F35078DB-6337-4339-AF5B-3A1FA5501A71/a-Shell.app/Frameworks/python3_ios.framework/python3_ios' (could not register fairplay decryption, mremap
_encrypted() => -1), '/usr/lib/python3_ios.framework/python3_ios' (errno=2, not in dyld cache), 'python3_ios.framework/python3_ios' (relative path n
ot allowed in hardened program), '/System/Library/Frameworks/python3_ios.framework/python3_ios' (errno=2, not in dyld cache)
python: command not found

The commands seem to change each time I start a-shell, python has started successfully on some occasions!
Happy to provide additional info if required.

Thanks

**Comment by holzschu** [comment]
Hi, 
thank you for reporting this. It is identical to issue #477. According to Apple support, this is a "weird issue" that should not happen. In short, from what the error message says, the binary for the command is there, but it was not authentified and thus cannot be loaded.

If you can, please create a sysdiagnose log immediately after the issue happens (see instructions below) and forward it to another_shell@icloud.com
I'll forward it to Apple support and check if they can understand what is creating the issue.

[sysdiagnose_Logging_Instructions.pdf](https://github.com/holzschu/a-shell/files/9945817/sysdiagnose_Logging_Instructions.pdf)

**Comment by arcane58** [comment]
Sorry, I did not check carefully enough to see if the issue had already been reported....

I have generated a sysdiagnose as per the instructions, however it is 250MB so couldn't send by email

Trying to attach here.........

Nope that doesn't work either as the limit seems to be 25MB

I have put the file on Dropbox and sent the link to your email, please let me know if you are able to access.

The sysdiagnose was triggered within 5 seconds of me reproducing the issue.

**Comment by holzschu** [comment]
Thanks a lot for the file. As you can see from #477, the issue seems to be solving itself without explanation. You are the first to be able to generate a sysdiagnose. I have downloaded, I hope it will contain enough information to fix the issue.

**Comment by arcane58** [comment]
Not sure if this adds anything to help, but yesterday afternoon, I installed a-shell mini alongside the original a-shell installation.
Although I have not spent hours trying things out, I cannot get a-shell mini to fail in the same way as a-shell.
I have also tried executing commands from a-shell in IOS shortcuts and again have not encountered the same errors as I get with the a-shell app itself.

**Comment by holzschu** [comment]
Yes, a-Shell mini is affected less frequently than a-Shell, but it still is. Re-installing the app has (often) solved the issue, as well as, apparently, waiting for a few days. All those things are pointing out towards an issue with the download and install process, rather than with the app itself. I hope Apple will be able to solve things.

**Comment by arcane58** [comment]
I can confirm that re-installing seems to resolve the issue, I re-installed a few days ago and although I have not tested as much as I would like to, I have not seen the issue recur since then.

**Comment by holzschu** [comment]
Thanks for confirming that. A question asked by Apple: was the issue happening immediately after download, or instead after a certain number of commands?

**Comment by arcane58** [comment]
It was immediately after download, that I had problems, the first time I attempted to do anything the window would not accept any input. After closing the app and restarting, it accepted input but gave the error on simple commands like "pwd" and "ls -l", but I could usually find a command that did work. But as I said previously, the commands that produced the error were not always the same, sometimes the app would let me do "pwd" and commands like "python" or "ssh" would fail with the error!

**Comment by holzschu** [comment]
Thank you very much. I've been told it will help them in eliminating possible causes for the bug.

**Comment by holzschu** [comment]
Update: I have filed an official bug report with Apple (FB11774729), using your sysdiagnose file (since the issue solves itself randomly, you're the only user who managed to produce a sysdiagnose). I'll keep you posted on their reply.
